### ISSUE TITLE
Storage running full

### ISSUE THREAD

**Issue by KevinWassermann94** [NebraLtd/hm-miner#29]
We need a better solution for wiping the storage before it is full. There has been hotspot storages getting full after the update.
For now the storage will only wipe on a reboot or an update if it is more than 80% full. 
It should check for the storage used at least once a day, as there might not be a reboot/ update before it is full after 80% is reached.

**Comment by vpetersson** [comment]
@KevinWassermann94 as things stand right now, there is no other way to do this. We can't proactively clean this up. The only avenue we have is what we have implemented. It sucks, but not much we can do about it.

**Comment by shawaj** [comment]
https://github.com/NebraLtd/hm-disk-purger

**Comment by vpetersson** [comment]
Yeah, so I've been toying with the idea of introducing a 'watchdog' container that will be responsible for various system tasks. As we are removing hm-gwmfr, we will free up some resources.

**Comment by shawaj** [comment]
that could be useful - presumably that could have access to balena supervisor and could also do the mdns?

**Comment by shawaj** [comment]
could even stick the upnp stuff in there perhaps too?

**Comment by vpetersson** [comment]
> that could be useful - presumably that could have access to balena supervisor and could also do the mdns?

Yeah the idea is basically to grant this Supervisor access but not exposing it publicly.

**Comment by shawaj** [comment]
https://github.com/NebraLtd/hm-watchdog for future reference
